### PR TITLE
MQE-1962: Remove support of deprecated locations for tests

### PR DESCRIPTION
**Pull request by jilu1**
### Fixed Issues (if relevant)

1. magento/magento2-functional-testing-framework#<issue_number>: Issue title
2. ...

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/verification tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)
 - [ ] Changes to Framework doesn't have backward incompatible changes for tests or have related Pull Request with fixes to tests

### CODE REVIEW

**Comment by soumyau** [comment]
Looks good, needs a green build with test changes attached to the ticket.

File: docs/mftf-tests-packaging.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+vendor/<Vendor>/<Module>/Test/Mftf
+vendor/<Vendor>/<TestModule>
+```
+
+To support future service isolation, Test module in `dev/tests/acceptance/tests/functional/<Vendor>/<TestModule>` and 
+`vendor/<Vendor>/<TestModule>` must define the module type as `magento2-functional-test-module` in a `composer.json` file. 
+No `composer.json` file is required for tests in `app/code/<Vendor>/<Module>/Test/Mftf` and `vendor/<Vendor>/<Module>/Test/Mftf` 
+as they are part of the Magento modules.
+
+Test module for a specific Magento module can only be in one of the path.
```
[the remark this answers is not in the document]
Comment (jilu1) [reply docs/mftf-tests-packaging.md]: Fixed

File: docs/mftf-tests-packaging.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+The Magento Functional Testing Framework can run tests from predefined paths and custom paths. The predefined paths are:
+```
+app/code/<Vendor>/<Module>/Test/Mftf
+dev/tests/acceptance/tests/functional/<Vendor>/<TestModule>
+vendor/<Vendor>/<Module>/Test/Mftf
+vendor/<Vendor>/<TestModule>
+```
+
+To support future service isolation, Test module in `dev/tests/acceptance/tests/functional/<Vendor>/<TestModule>` and 
+`vendor/<Vendor>/<TestModule>` must define the module type as `magento2-functional-test-module` in a `composer.json` file.
```
[the remark this answers is not in the document]
Comment (jilu1) [reply docs/mftf-tests-packaging.md]: Fixed

**Comment by jilu1** [comment]
@soumyau There were merge conflicts and could you please re-review? Thanks!

**Pull request merged**